### DISCUSSION TITLE
Should I change /dev/ttyAMC0 to a "by-id" name?

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js-ui#3015] [Q&A]
I've been told that in order to avoid potential problems with HAOS reassigning USB ports during restarts, or after power loss, etc., that I should be using "by-id" port names, not the HW names. First, is this true? I've not had a problem to date with this to date. And, second, can I just retrieve this from the All Hardware info in HA and replace it in the Settings in ZWaveJS-UI (or would that be bad)?

This came up as I was asking about restoring my HA installation from a failed SD Card, by using a fresh HAOS flash on a new card and either going through the onboarding and then restoring a backup, or, if I can get the backup on _before_ onboarding (SSH?), go directly to restoring from a backup. In either case, I am expecting that my Zwave network would come back up as it was (same stick, HAOS/HA config as before), and really hope that is the case. Old threads (elsewhere) indicate that all the node names, etc., were changed under such circumstances. I want to avoid that at all cost, so I'm just trying to get ahead of it by preparing and knowing exactly _what_ to do _when_ it happens.

Any advice would be appreciated.

**Comment by robertsLando** [comment]
> I've been told that in order to avoid potential problems with HAOS reassigning USB ports during restarts, or after power loss, etc., that I should be using "by-id" port names, not the HW names. First, is this true?

Yeah that's true, the reason is that when driver starts it performs soft resets that may cause issues if the usb path used is not 'static' (the /dev/tty path are alias path that could change.

> And, second, can I just retrieve this from the All Hardware info in HA and replace it in the Settings in ZWaveJS-UI (or would that be bad)?

I cannot answer to this as I don't use HA, btw I added a guide [here](https://github.com/zwave-js/zwave-js-ui/discussions/2483) to discover it

> Old threads (elsewhere) indicate that all the node names, etc., were changed under such circumstances. I want to avoid that at all cost, so I'm just trying to get ahead of it by preparing and knowing exactly what to do when it happens.

In order to be 100% sure nothing changes or at least that you can always recover you have to do a backup of the store directory and also an nvm backup, even better if you enable automatic backup from settings. Store backups backup all the file needed like settings.json, nodes.json, scenes.json, users.json and all driver jsonl files. You can do a store backup directly from store explorer by pressing on the floating action button and pressing on backup. For the nvm backup just go to control panel, advanced actions, nvm backup.

Once you have both backups you can go on, at least for what is concerning zwave side you are ok. For hass side I dunno if you have to backup something but for sure @kpine knows.

Let me know how this goes

**Reply by GrizzlyAK** [reply]
Thanks for the reply, Daniel. Couple questions:

> I cannot answer to this as I don't use HA

In HA, I can find HW related data like that shown below:
![image](https://user-images.githubusercontent.com/4411963/227340929-7f31949f-0738-4bf2-88dc-fbf728310761.png)

In that is the "by-id" string we need, and what your link also discovers. My question was really about whether it is OK in zwavejs-ui, in the settings (as shown below) to simply change that setting to be the "by-id" string instead of the /dev/tty/ACM0. I have the string I need, and know where to change it, I just want to confirm that I won't break something in ZW if I change it in settings there.

![image](https://user-images.githubusercontent.com/4411963/227341786-ed90e0c9-900b-42bd-9c76-0bf132054a47.png)


>Once you have both backups you can go on,

Good idea on the backup. I have them set to automatically, using the settings below:

![image](https://user-images.githubusercontent.com/4411963/227342439-9df2a4c8-e6c3-4167-aaeb-74f100a0a568.png)

Looks like I do them weekly, keeping 7 copies. I confirmed in the store.

Finally, once I have the backups, the most likely case in which I will need them is if/when my SD Card fails completely and I have to restore HAOS using a _new_ install, and then relying on backups to restore to operation. I'm also working through currently how to do that for HA itself. In that case, I need to store these zwave backups OFF my HA disk and on my PC. So I can just download these from the `backups` folder in store, right? In order to get these back to a new HA disk to restore, I'll at least need a running zwavejs-ui, so that means restoring that first. My concern is if going through that HA restore (which, I assume, includes the zwave-ui add-on itself - but not the Store data?), is something lost that can't be recovered when finally doing a restore in zwavejs-ui. One thing that springs to mind is if a new zwvejs-ui install might wipe my Z-Stick controller, or rename everything, and if so, will a restore of the zw Store/NVM restore all of that (or will I have to rename everything). Speaking of, there is a Backup section, but I don't see a Restore section. I guess I need to go read the Documentation again. It's been a while.

Cheers.

BTW, after much research, I finally just relented and gave GH one of my seldom-used CCs so that if they lose it, it won't be too painful. I also upped your sponsorship, and restored @AlCalzone 's too. Did that yesterday.

**Reply by AlCalzone** [reply]
node-zwave-js only uses the serial path to know which port to open. All identification uses the actual communication with the stick behind it. As long as that's the same, you should be able to change ports and paths as much as you like.

**Reply by GrizzlyAK** [reply]
Thanks! When will this take effect (or when will I know if it doesn't work)? When I restart the zwavejs-ui add-on? Does that restart the driver too, or is that now part of HA core? Guess I'll find out. I just changed it and restarted the add-on. 🤣

**Reply by AlCalzone** [reply]
saving the settings will cause Z-Wave JS UI to restart the driver

**Reply by robertsLando** [reply]
> My question was really about whether it is OK in zwavejs-ui, in the settings (as shown below) to simply change that setting to be the "by-id" string instead of the /dev/tty/ACM0

Like @AlCalzone said that is just the path the driver uses to open the serial port so even if you input it wrongly nothing is lost, you will just see errors that the port cannot be opened but once you input it correctly it just works as before.

>  So I can just download these from the backups folder in store, right?

Yes, right!

>  One thing that springs to mind is if a new zwvejs-ui install might wipe my Z-Stick controller, or rename everything, and if so, will a restore of the zw Store/NVM restore all of that (or will I have to rename everything). Speaking of, there is a Backup section, but I don't see a Restore section. I guess I need to go read the Documentation again. It's been a while.

You can restore a backup directly from store ui, just press on the floating action button and select restore, it will prompt you to select the desired restore file and it will do the rest. This is for the store restore, for the nvm restore you have to do that from control panel > advanced actions > restore nvm.

And yeah in case something wrong happens you just have to do a clean install of zui, open the ui, do the store restore and, if you replaced your stick for some reason, also do the NVm restore. But if your stick is the same you don't have to restore the nvm, that is just to recover stick memory

> BTW, after much research, I finally just relented and gave GH one of my seldom-used CCs so that if they lose it, it won't be too painful. I also upped your sponsorship, and restored @AlCalzone 's too. Did that yesterday.

Saw that yesterday, thanks so much for that @GrizzlyAK  🙏🏼 ❤️

**Reply by GrizzlyAK** [reply]
OK, all that makes me feel a lot more prepared in case of trouble. Thanks guys! Keep up the great work. Cheers.

**Reply by AlCalzone** [reply]
Oh, thanks a lot for the updated sponsorship! Were the alternatives that accept PayPal not an option?

**Reply by GrizzlyAK** [reply]
I looked at all the options, but there was a fee that took away from the contribution. Patreon was the absolute worst. Yikes. Daniel indicated that GitHub takes nothing, and I assumed that meant no CC processing fees either (might be wrong), but that seemed like the  best option to get the most to you guys without a middle man getting a cut.
